helm/cluster/templates/apps/apps.yaml

```diff
@@ -0,0 +1,141 @@
+{{- $rootScope := . }}
+{{- range $path, $_ := .Files.Glob "files/apps/**.yaml" }}
+{{- $app := dict }}
+{{- with $rootScope }}
+{{- $app = (tpl (.Files.Get $path) $) | fromYaml }}
+{{- $appName := $app.appName }}
+{{- $appKey := $app.configKey }}
```

Comment:
Do we need a separate `configKey`? Could we instead use `appName` as we can then easily work this out in testing from an apps own PR (for example).
If I understand correctly the only reason for using a `configKey` is to allow the name to not have hyphens in it but as we are only using it with the `get` calls below it shouldn't make a difference, right?

Otherwise, we need some way of building the `configMap` based on the app repo.

Reply:
`configKey` is the name of the app config entry in these Helm values:
- `.Values.global.apps`,
- `.Values.providerIntegration.apps` and in `.Values.internal.ephemeralConfiguration.apps`

so the key is meant to be predefined/known API with naming aligned with all other Helm values (camelCase here), especially for `.Values.global.apps` which a public API for customers.

> Otherwise, we need some way of building the configMap based on the app repo.

Where and why you need this?

Do you have some case where you dynamically/programatically build app configs?